### PR TITLE
minor: Add a short blurb about VS Code Live Share to the manual

### PR DESCRIPTION
**Pull request by jonas-schievink**
Part of https://github.com/rust-lang/rust-analyzer/issues/8844

### CODE REVIEW

File: docs/user/manual.adoc [hunk 1]
```diff
@@ -861,3 +861,14 @@ For example, if you want to run https://crates.io/crates/cargo-watch[`cargo watc
...
 }
 ```
+
+==== Live Share
+
+VS Code Live Share has partial support for rust-analyzer.
+
+Live Share _requires_ the official Microsoft build of VS Code, OSS builds will not work correctly.
+
+The host's rust-analyzer instance will be shared with all guests joining the session.
```
[the remark this answers is not in the document]
Comment (jonas-schievink) [reply docs/user/manual.adoc:871]: No, there are quite a few limitations right now:

- no inlay hints
- no snippet completions / flyimport (the import doesn't get injected)
- no editor commands work, which means that anything that invokes a command also doesn't work:
  - extend/shrink selection
  - move item
  - ...
- links in hover popups don't work

But at least assists, completion, runnables, goto def, etc. work, so it's not "Rust in 2015" bad

[the remark this answers is not in the document]
Comment (DrChat) [reply docs/user/manual.adoc:871]: Out of curiosity, what's preventing inlay hints from working in a live share session?
Does rust-analyzer need some work to add support for that?

[the remark this answers is not in the document]
Comment (jonas-schievink) [reply docs/user/manual.adoc:871]: No, I think Live Share just doesn't support them yet

**Pull request merged**